mods/Fifty.ServerAdmin/mod/scripts/vscripts/fsa_core.nut

```diff
@@ -126,6 +190,239 @@ bool function FSA_IsAdmin( entity player ) {
 
 	return false
 }
+/*
+ * Returns the entity of a player name returns null if no player was found
+ * @ param name The name of the player you want the entity of
+*/
+entity function FSA_GetPlayerEntityByName(string name){
+
+	if(name == "")
+    	return null
+
+
+  foreach(entity p in GetPlayerArray())
+    if(p.GetPlayerName().tolower() == name.tolower())
+      return p
+
+
+  if(name.len() <= 2){
+    int PlayerIndex = name.tointeger()
+    if(GetPlayerArray().len()-1>= PlayerIndex && PlayerIndex > -1) // check if the index exits
+      return GetPlayerArray()[PlayerIndex]
+  }
+
+  return null
+}
+
+/**
+ * Gets called when a player runs !playerlist
+ * @param player The player who caled the command
+ * @param args The arguments passed by the player
+*/
+void function FSCC_CommandCallback_Playerlist( entity player, array< string > args ){
+	array< string > printList
+	foreach( index, entity p in GetPlayerArray() ) {
+		printList.append( p.GetPlayerName() )
+		printList.append( index.tostring() )
+	}
+	int columns = 2
+	int pages = FSU_GetListPages(printList, columns) +1
+	for( int page = 1; page< pages; page++ )
+		FSU_PrintFormattedList(player, printList, page, columns)
+}
+
+/**
+ * Gets called when a player runs !script
+ * @param player The player who caled the command
+ * @param args The arguments passed by the player
+*/
+void function FSCC_CommandCallback_Script(entity player, array<string> args){
+  if(args.len() == 0){
+    FSU_PrivateChatMessage(player, "Missing arguments: !script <code here>")
+    return
+  }
+  try{
+    compilestring( FSU_ArrayToString(args) )()
+    FSU_PrivateChatMessage(player, "Your code seems to have compiled")
+    return
+  }
+  catch ( ex ){
+    printt(ex)
+    FSU_PrivateChatMessage(player, "The code has caused an exception, the error can be found in the server log")
+    return
+  }
+}
+
+/**
+ * Gets called when a player runs !serverCommand
+ * @param player The player who caled the command
+ * @param args The arguments passed by the player
+*/
+void function FSCC_CommandCallback_ServerCommand(entity player, array<string> args){
+  if(args.len()==0)
+  {
+    FSU_PrivateChatMessage(player, "Missing arguments")
+    return
+  }
+  try{
+    ServerCommand(FSU_ArrayToString(args))
+  }
+  catch(ex){
+    FSU_PrivateChatMessage(player,"The command has caused an exception")
+  }
+  FSU_PrivateChatMessage(player, "Command executed")
+
+}
+
+/**
+ * Gets called when a player runs !reload
+ * @param player The player who caled the command
+ * @param args The arguments passed by the player
+*/
+void function FSCC_CommandCallback_Reload(entity player, array<string> args){
+
+  if(args.len() <= 0)
+    thread FSU_C_Reload_thread(5.0)
+  else
+    thread FSU_C_Reload_thread(args[0].tofloat())
+}
+
+void function FSU_C_Reload_thread(float time){
+  while(time > 0){
+    Chat_ServerBroadcast("The server will reload in "+ time)

```

Comment:
Use `FSU_ChatBroadcast`

Reply:
what about the RUI messages do I swap them too?

Reply:
okay swaped all of them